plugins/powerd/fu-plugin-powerd.c

```diff
@@ -20,6 +20,41 @@ fu_plugin_init(FuPlugin *plugin)
 	fu_plugin_alloc_data(plugin, sizeof(FuPluginData));
 }
 
+static gboolean
+fu_plugin_powerd_create_suspend_file(GError **error)
+{
+	g_autofree gchar *lockdir = NULL;
+	g_autofree gchar *inhibitsuspend_filename = NULL;
+	g_autofree gchar *getpid_str = NULL;
+
+	lockdir = fu_common_get_path(FU_PATH_KIND_LOCKDIR);
+	inhibitsuspend_filename = g_build_filename(lockdir, "power_override", "fwupd.lock", NULL);
+	getpid_str = g_strdup_printf("%d", getpid());
+	if (!g_file_set_contents(inhibitsuspend_filename, getpid_str, -1, error)) {
+		g_prefix_error(error, "lock file unable to be created");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_powerd_delete_suspend_file(GError **error)
+{
+	g_autoptr(GError) local_error = NULL;
+	g_autofree gchar *lockdir = NULL;
+	g_autoptr(GFile) inhibitsuspend_file = NULL;
+
+	lockdir = fu_common_get_path(FU_PATH_KIND_LOCKDIR);
+	inhibitsuspend_file =
+	    g_file_new_build_filename(lockdir, "power_override", "fwupd.lock", NULL);
+	if (!g_file_delete(inhibitsuspend_file, NULL, &local_error) &&
+	    !g_error_matches(local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND)) {
+		g_prefix_error(error, "%s", local_error->message);
```

Comment:
g_propagate_prefixed_error(error, local_error, "lock file unable to be deleted");

Reply:
Actually I think you need a steal pointer: 
		g_propagate_prefixed_error(error, g_steal_pointer(&error_local), "lock file unable to be deleted");